h2/utilities.py

```diff
@@ -484,7 +480,7 @@ def validate_outbound_headers(headers, hdr_validation_flags):
     headers = _reject_te(
         headers, hdr_validation_flags
     )
-    headers = _reject_connection_header(
+    headers = _strip_connection_header(
```

Comment:
Because this is modifying the outbound headers rather than checking for correctness, I think it should be a step in `normalize_outbound_headers` instead.

Implicit in `validate_outbound_headers` is that you either get a `ProtocolError` or the headers returned unmodified.